### DISCUSSION TITLE
useFieldArray with useFormContext doesn't work properly

### DISCUSSION THREAD

**Discussion by niteshgrg** [react-hook-form/react-hook-form#5672] [Q&A] [answered]
**Describe the bug**
I looked into previous issue as well (https://github.com/react-hook-form/react-hook-form/issues/1198) but for me i am using controllers for inputs so cannot use register(). 

**To Reproduce**
Easily reproducible with any formContext and fields with controller. Sorry if there is any clear cut syntax on how to use controllers with context and i am missing it.

Steps to reproduce the behaviour:
1. Create a fieldArray from formContext and inputs registered with Register input with controller
2. Add few fields with append() and remove with remove()
3. Submit data it becomes empty.

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-usefieldarray-forked-3zd6g?file=/src/index.js

**Comment by bluebill1049** [comment, marked as answer]
What's the issue here? Can u elaborate the expectation? If I append once and delete, obviously the date is going to be empty.

**Reply by niteshgrg** [reply]
@bluebill1049 you can add 3 times and delete 1 time its still empty. Basically delete operations makes everything in that field array to not come in submit. I saw this issue being tracked in other issue as well. Solution was to use register() but we cannot do that in controller.

**Reply by niteshgrg** [reply]
For steps 

- add 3 fields 
- Enter data into all fields
- delete one field
- Try submit now
- See data console.

**Reply by bluebill1049** [reply]
<img width="418" alt="Screen Shot 2021-06-18 at 4 33 16 pm" src="https://user-images.githubusercontent.com/10513364/122517052-e3a3f180-d052-11eb-8c24-b08ae5a76e98.png">

we do no support flat array.

**Reply by niteshgrg** [reply]
Ok thanks for the answer.
